dtls.c

```diff
@@ -4061,15 +4062,30 @@ dtls_handle_message(dtls_context_t *ctx,
       dtls_security_parameters_t *security = dtls_security_params_epoch(peer, dtls_get_epoch(header));
       if (!security) {
         dtls_alert("No security context for epoch: %i\n", dtls_get_epoch(header));
-        data_length = -1;
+        return -1;
       } else {
+        uint16_t msg_epoch = dtls_get_epoch(header);
+
+        if (peer->handshake_params &&
+            peer->handshake_params->hs_epoch != msg_epoch &&
+            peer->state != DTLS_STATE_WAIT_FINISHED) {
+          if ((uint16_t)(peer->handshake_params->hs_epoch + 1) == msg_epoch) {
```

Comment:
I do not see why the type cast is necessary here.

Reply:
I added it in in case `hs_epoch `reached 0xffff (very unlikely) and the +1 caused it not to correctly wrap.

Reply:
I agree, in that case the `==` would yield different results. (Notwithstanding the fact that this must not happen in a correct DTLS implementation.)